google-cloud-workflows/src/test/java/com/google/cloud/workflows/v1beta/it/ITSystemTest.java

```diff
@@ -0,0 +1,126 @@
+          + "   #   the response in CurrentDateTime variable  "
+          + "   # - retrieves a list of Wikipedia articles related to the day of the week  "
+          + "   #   from CurrentDateTime  "
+          + "   # - returns the list of articles as an output of the workflow  "
+          + "     "
+          + "   - getCurrentTime:  "
+          + "       call: http.get  "
+          + "       args:  "
+          + "           url: https://us-central1-workflowsample.cloudfunctions.net/datetime  "
+          + "       result: CurrentDateTime  "
+          + "   - readWikipedia:  "
+          + "       call: http.get  "
+          + "       args:  "
+          + "           url: https://en.wikipedia.org/w/api.php  "
+          + "           query:  "
+          + "               action: opensearch  "
+          + "               search: ${CurrentDateTime.body.dayOfTheWeek}  "
+          + "       result: WikiResult  "
+          + "   - returnOutput:  "
+          + "      return: ${WikiResult.body[1]}  ";
+  private static WorkflowsClient client;
+  private static Workflow expectedWorkflow;
+
+  @BeforeClass
+  public static void setUp() throws IOException, ExecutionException, InterruptedException {
+    // Create Test Workflow
+    client = WorkflowsClient.create();
+    Workflow workflow =
+        Workflow.newBuilder().setName(WORKFLOW_NAME).setSourceContents(SOURCE_CONTENT).build();
+    CreateWorkflowRequest request =
+        CreateWorkflowRequest.newBuilder()
+            .setParent(PARENT)
+            .setWorkflow(workflow)
+            .setWorkflowId(WORKFLOW_ID)
+            .build();
+    expectedWorkflow = client.createWorkflowAsync(request).get();
+  }
+
+  @AfterClass
+  public static void tearDown() {
+    // Delete Test Worflow
+    DeleteWorkflowRequest deleteRequest =
+        DeleteWorkflowRequest.newBuilder().setName(WORKFLOW_NAME).build();
+    client.deleteWorkflowAsync(deleteRequest);
+    client.close();
+  }
+
+  @Test
+  public void testGetWorkflow() {
+    Workflow workflow = client.getWorkflow(WORKFLOW_NAME);
+    assertEquals(expectedWorkflow, workflow);
+  }
+
+  @Test
+  public void testUpdateWorkflow() throws ExecutionException, InterruptedException {
+    Workflow workflow =
+        Workflow.newBuilder().setName(WORKFLOW_NAME).setDescription(WORKFLOW_DESCRIPTION).build();
+    FieldMask updateField = FieldMaskUtil.fromString("description");
+    UpdateWorkflowRequest request =
+        UpdateWorkflowRequest.newBuilder().setWorkflow(workflow).setUpdateMask(updateField).build();
+    Workflow updateWorkflow = client.updateWorkflowAsync(request).get();
+    assertEquals(expectedWorkflow, updateWorkflow);
+  }
+
+  @Test
+  public void testListWorkflows() {
+    ListWorkflowsRequest request = ListWorkflowsRequest.newBuilder().setParent(PARENT).build();
+    for (Workflow actualWorkflow : client.listWorkflows(request).iterateAll()) {
+      assertEquals(WORKFLOW_NAME, actualWorkflow.getName());
```

Comment:
nit : this checks will failed in case of system already has workflow

better to put check here with workflowname like:

 ```javascript
for (Workflow actualWorkflow : client.listWorkflows(request).iterateAll()) {
       if(WORKFLOW_NAME.equals(actualWorkflow.getName()){
             assertEquals(WORKFLOW_NAME, actualWorkflow.getName());
             assertEquals(expectedWorkflow, actualWorkflow);
     }
}
```